src/dsp_tools/commands/ingest_xmlupload/apply_ingest.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import glob
+from typing import cast
+
+import pandas as pd
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.user_information import IngestInformation
+from dsp_tools.models.exceptions import InputError
+
+
+def get_mapping_dict_from_file() -> dict[str, str]:
+    """
+    This functions returns the information to replace the original filepaths with the identifier from dsp-ingest.
+
+    Returns:
+        dictionary with original: identifier from dsp-ingest
+
+    Raises:
+        InputError: if no file was found
+    """
+    if filepath := glob.glob("mapping-*.csv"):
+        df = pd.read_csv(filepath[0])
+        print(f"The file '{filepath[0]}' is used to map the internal SIPI image IDs to the original filepaths.")
+        return dict(zip(df["original"].tolist(), df["derivative"].tolist()))
+    else:
+        raise InputError("No mapping csv file was found in the directory.")
+
+
+def replace_filepath_with_sipi_uuid(
+    xml_tree: etree._ElementTree[etree._Element],
+    orig_path_2_uuid_filename: dict[str, str],
+) -> tuple[etree._ElementTree[etree._Element], IngestInformation]:
+    """
+    Replace the original filepaths in the <bitstream> tags by the uuid filenames of the uploaded files.
+
+    Args:
+        xml_tree: The parsed original XML tree
+        orig_path_2_uuid_filename: Mapping from original filenames to uuid filenames from the mapping.csv
+
+    Returns:
+        The XML tree with the replaced filepaths (modified in place)
```

Comment:
You may want to have a look at my comment above: In the long run, we should not modify the XML at all, but rather we should think of DSP-TOOLS as a data pipeline which applies a series of transformations to the data that flows through. Ideally, this would just be a transformation step by itself, which would then render this entire question obsolete. And for that reason I'm not sure we have to necessarily follow this kind of best practice here.  
But yes, in principle I think you're right.

Reply:
Hmmm... You are speaking in riddles... What comment above? I don't find it.

Reply:
sorry, yes, in file view and before it was marked as resolved, it was easier to see ^^  
here you go: https://github.com/dasch-swiss/dsp-tools/pull/670/files/f2ed9e9877d78287ad5c2984e319baf4176216b2#r1421403136